### PR TITLE
Configure PyGlslang builds

### LINKED ISSUE #1: Build glslang on ubuntu
Build [glslang](https://github.com/KhronosGroup/glslang/) on Ubuntu following guides from [their readme](https://github.com/KhronosGroup/glslang/tree/main#building-cmake).

### LINKED ISSUE #2: Automate glslang builds & packaging with poetry and `build.py`
To correctly package glslang into wheel and sdist it is necessary to configure project in way understandable for pip and other package managers. You can refer to how [cmake](https://github.com/scikit-build/cmake-python-distributions) PyPI distribution is done or to any other resources out there. As a result a single command (`pip install pyglslang`) should be sufficient to build pyglslang from source on Windows. I assume that this program will be invoked via CLI, so we can just compile it to platform specific executable and copy inside package afterwards to embed it into wheel.

### PR DESCRIPTION
**Pull request by MatSzpy**
Resolves: #1
Resolves: #2

### CODE REVIEW

**Review by Argmaster — DISMISSED** [review]

File: src/pyglslang/api.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import subprocess
+from test.test_glsl import test_glsl
+
+
+def compile_shader(options: CompilerOptions) -> CompilerResults:
+    """Compile shader."""
```
Comment (Argmaster) [inline src/pyglslang/api.py:7]: iuy9fxt300811

**Review by Ave44 — DISMISSED** [review]

**Review by mateuszzalewski011 — DISMISSED** [review]

File: src/pyglslang/api.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import subprocess
+from test.test_glsl import test_glsl
+import tempfile
+
+
+def compile_shader(options: CompilerOptions) -> CompilerResults:
+    """Compile shader."""
+    params = options.get_executable_params()
+    result = subprocess.run(params, capture_output=True)
+    temp = tempfile.NamedTemporaryFile(name="temp_result", dir="~/pyglslang/src/pyglslang")
```
Comment (Argmaster) [inline src/pyglslang/api.py:10]: Leaving `~` in here may not work on windows, I also do not understand why are you specifying non-default directory for this temporary file?

To be honest I don't understand this code at all. Why are you storing stdout of glslang but you are neither specifying output file path, nor providing some known cwd for glslang to access output file. 
And of course you are not reading compiled SPIR-V from file created by glslang.

File: src/pyglslang/compiler_options.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
+from typing import Optional
+
+class CompilerOptions:
+    def __init__(self,first_glslang_option: Optional[bool]) -> None:
+        pass
+    def get_executable_params(self) -> list[str]:
+        """Get parameters that will be executed."""
+        params: list[str] = ["glslang"]
```
Comment (Argmaster) [inline src/pyglslang/compiler_options.py:8]: I'd recomment removing `glslang` from here, so the list contains only params themselves, glslang executable path can be added later on.

File: src/pyglslang/compiler_results.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        pass
+    def get_path(self) -> Path:
+        """Get path to compiled source file."""
+        path = Path(temp_result).parent
+        return path
+    def get_return_code(self) -> int:
+        """Get glslang return code."""
+    def good() -> bool:
+        """Check if compilation didn't fail."""
+        test = subprocess.check_output(test_glsl)
```
Comment (Argmaster) [inline src/pyglslang/compiler_results.py:15]: From python documentation for `subprocess.check_output()`

> Run command with arguments and return its output.

I don't think it is what you want. You should be storing CompletedProcess object returned by `subprocess.run` which invoked `glslang` instead. Then you'd be able to check if `result.returncode` is equal to `0`. Programs tend to return `0` when they succeed and non-zero code when they fail.

File: src/pyglslang/api.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import subprocess
+from test.test_glsl import test_glsl
+import tempfile
+
+
+def compile_shader(options: CompilerOptions) -> CompilerResults:
+    """Compile shader."""
+    params = options.get_executable_params()
+    result = subprocess.run(params, capture_output=True)
```
Comment (Argmaster) [inline src/pyglslang/api.py:9]:
```suggestion
    glslang_executable_path = get_glslang_executable_path()
    result = subprocess.run([glslang_executable_path, *params], executable=glslang_executable_path, capture_output=True)
```

**Review by Argmaster — CHANGES_REQUESTED** [review]

**Pull request closed without merging**